lib/deploy.bash

```diff
@@ -0,0 +1,428 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# shellcheck source=lib/plugin.bash
+. "${DIR}/plugin.bash"
+
+function deploy_lambda() {
+  local deployment_method="${BUILDKITE_PLUGIN_AWS_LAMBDA_DEPLOY_DEPLOYMENT_METHOD:-aws-cli}"
+
+  if [[ "${deployment_method}" == "sam" ]]; then
+    # For SAM, we don't pass the function/alias name directly, it's in the template
+    deploy_sam
```

Comment:
This doesn't seem to be defined anywhere.

Reply:
Oops, that bit wasn't supposed to be committed in this branch. I've been experimenting with [AWS SAM](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/using-sam-cli.html) as an alternative deployment method.